src/modal/modal.scss

```diff
@@ -71,6 +73,72 @@
   &.iui-modal-visible {
     visibility: visible;
     opacity: 1;
-    transition-delay: 0s;
+    transition-delay: $iui-speed-instant;
+  }
+}
+
+@mixin iui-modal-full-page {
+  > .iui-modal-dialog {
+    display: flex;
+    flex-direction: column;
+    height: 100vh;
+    width: 100vw;
+    left: 0;
+    top: 0;
+    transform: none;
+    max-width: initial;
+    min-width: initial;
+    border-radius: 0;
+  }
+
+  .iui-modal-content {
```

Comment:
I think you need to move this out of full-modal because regular modal users could want to use it and also in React I think we will create separate component for this one.